### ISSUE TITLE
v1.2.1 introduces a bug with HTTP event Request Parameters

### ISSUE THREAD

**Issue by pas256** [serverless/serverless#2798]
# This is a Bug Report

## Description

For bug reports:
* What went wrong?

When doing a `sls deploy`, the following error occurs:

```
............................Serverless: Deployment failed!

  Serverless Error ---------------------------------------

     An error occurred while provisioning your stack: ApiGatewayMethodBroadcastIdVarGet
     - Invalid mapping expression specified: Validation Result:
     warnings : [], errors : [Invalid mapping expression
     specified: paths].

  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues

  Your Environment Information -----------------------------
     OS:                 darwin
     Node Version:       6.5.0
     Serverless Version: 1.2.0
```

* What did you expect should have happened?

When I use Serverless 1.1.0, it works as expected.

* What was the config you used?

The relavant part of the config file is:

```yaml
functions:
  http:
    name: ${self:custom.${env:PROG_ENV}.name.lower}-http
    handler: handlers.http_handler
    timeout: 5
    events:
      - http:
          method: get
          path: resource/{id}
          request:
            parameters:
              paths:
                id: true
          cors:
            origins:
              - '*'
            headers:
              - Content-Type
              - Authorization
              - Accept
              - X-Amz-Date
              - X-Api-Key
              - X-Amz-Security-Token
              - x-secret
```

* What stacktrace or error message from your provider did you see?

    Invalid mapping expression specified: paths

## Additional Data

I also tested this with v1.2.0 and the issue happens there too. So I am going to stick with 1.1.0 until this get resolved.

If there is anything I can do to help, please let me know.

**Comment by dougmoscrop** [comment]
I've found the cause.

When you do an `sls deploy --noDeploy` I expect you should see:

```
Serverless: Warning! You're using the LAMBDA-PROXY in combination with request / response configuration in your function "http". This configuration will be ignored during deployment.
```

It's not actually 'ignoring' the configuration, but what it is doing is *not* parsing the configuration. So the un-transformed "parameters" object is being passed through

So if you

`cat .serverless/cloudformation-template-update-stack.json` you should see:

```json
"ApiGatewayMethodResourceIdVarGet": {
      "Type": "AWS::ApiGateway::Method",
      "Properties": {
        "HttpMethod": "GET",
        "RequestParameters": {
          "paths": {
            "id": true
          }
        }
  ...
```

This is the relevant-ish line of code: https://github.com/serverless/serverless/blob/master/lib/plugins/aws/deploy/compile/events/apiGateway/lib/validate.js#L72

So, I need @pmuens or someone to advise on what's supposed to happen here. The configuration that @pas256 has in their serverless.yml is said to be ignored, so we could just `delete http.request` (and `delete http.response`) - but this issue sort of implies people are ignoring that warning and expecting that configuration to set up api gateway for some reason, even though they're using proxy integration and I suspect that there might be value in that.

So, TL;DR is: I introduced a bug, it's a two line fix, but I'm not sure that once fixed, users will be getting the results they expect.

**Comment by pas256** [comment]
Thats for the quick response!

I changed by config to be:

```yaml
functions:
  http:
    name: ${self:custom.${env:PROG_ENV}.name.lower}-http
    handler: handlers.http_handler
    timeout: 5
    events:
      - http:
          method: get
          path: resource/{id}
          cors:
            origins:
              - '*'
            headers:
              - Content-Type
              - Authorization
              - Accept
              - X-Amz-Date
              - X-Api-Key
              - X-Amz-Security-Token
              - x-secret
```

... and now everything seems to be peachy.

**Comment by pmuens** [comment]
@pas256 thanks for finding this bug and thanks @dougmoscrop for providing a fix 👍  🎉 

Will look into it ASAP so that we can merge it soon!
